### PR TITLE
udev,pid1: gracefully handle failure in renaming network interface

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/udev/udev-event.c [hunk 1]
```diff
@@ -917,63 +918,97 @@ static int rename_netif(UdevEvent *event) {
...
+        (void) device_add_property(event->dev_db_clone, "ID_RENAMING", NULL);
+        (void) device_update_db(event->dev_db_clone);
+
+        /* Restore 'dev' */
+        (void) device_set_syspath(dev, old_syspath, /* verify = */ false);
+        if (sd_device_get_property_value(dev, "INTERFACE_OLD", &s) >= 0) {
+                (void) device_add_property_internal(dev, "INTERFACE", s);
+                (void) device_add_property_internal(dev, "INTERFACE_OLD", NULL);
+        }
+        (void) device_add_property(dev, "ID_RENAMING", NULL);
```
Comment (daandemeyer) [inline src/udev/udev-event.c:1009]: We don't need to restore the old sysname here?

**Review by DaanDeMeyer — APPROVED** [review]

File: src/udev/udev-event.c (hunk 1 above)
Comment (yuwata) [reply src/udev/udev-event.c:1009]: You mean "INTERFACE_OLD"?? INTERFACE_OLD is not set by kernel iirc, and only set by udevd. Hence, it is not necessary to restore INTERFACE_OLD.

Or, you asked about something like `device_set_sysname()`? `sysname` is generated from syspath when requested, so, restoring only syspath is enough.

File: src/udev/udev-event.c [hunk 2]
```diff
@@ -917,63 +918,97 @@ static int rename_netif(UdevEvent *event) {
...
-        if (r < 0)
-                return log_device_error_errno(dev, r, "Failed to rename network interface %i from '%s' to '%s': %m",
-                                              ifindex, oldname, event->name);
-
-        log_device_debug(dev, "Network interface %i is renamed from '%s' to '%s'", ifindex, oldname, event->name);
 
+        log_device_debug(dev, "Network interface %i is renamed from '%s' to '%s'", ifindex, old_sysname, event->name);
         return 1;
+
+fail:
```
Comment (poettering) [inline src/udev/udev-event.c]: maybe call this label "revert" rather than "fail" given it is also called with `r == 0`

**Comment by poettering** [comment]
lgtm

**Comment by yuwata** [comment]
@DaanDeMeyer and @poettering Thank you for your review. The above suggestion by @poettering is addressed. Upgrading the green label.

**Pull request merged**